### PR TITLE
Fix CI pipeline

### PR DESCRIPTION
**Pull request by urso**
- Fix

## What does this PR do?

Add missing updates from `make update`.

## Why is it important?

Try to fix lint step in master and PRs targeting master.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -204,8 +204,8 @@ def runLinting() {
   }
   mapParallelTasks['default'] = {
                                 cmd(label: "make check-python", script: "make check-python")
-                                cmd(label: "make check-go", script: "make check-go")
                                 cmd(label: "make notice", script: "make notice")
+                                cmd(label: "make check-go", script: "make check-go")
                                 cmd(label: "Check for changes", script: "make check-no-changes")
```
Comment (urso) [inline Jenkinsfile:209]: Update order to match `make check`. The `make notice` command can download dependencies, which results in `go.sum` being updated. The `make check-go` seems to clean the `go.sum` again. Depending on order of `make notice` and `make check-go` the lint checks are currently not satisfiable without requiring a fixed order

File: Makefile [hunk 2]
```diff
@@ -110,6 +110,7 @@ check-go:
 ## ccheck-no-changes : Check there is no local changes.
 .PHONY: check-no-changes
 check-no-changes:
+	@go mod tidy
```
Comment (urso) [inline Makefile:113]: Per Beats lint runs `make check; make update; make check-no-changes`. The `make update` call touches `go.sum` when indirectly calling `make notice`. Try to be more aggressive before checking and clean up intermediary updates to `go.sum` before checking via `git diff`.

**Comment by andresrc** [comment]
The linting part seems to pass, let's re-run the tests

File: Jenkinsfile [hunk 3]
```diff
@@ -204,8 +204,8 @@ def runLinting() {
   }
   mapParallelTasks['default'] = {
                                 cmd(label: "make check-python", script: "make check-python")
-                                cmd(label: "make check-go", script: "make check-go")
                                 cmd(label: "make notice", script: "make notice")
+                                cmd(label: "make check-go", script: "make check-go")
```
Comment (cachedout) [inline Jenkinsfile:208]:
```suggestion
                                // `make check-go` must follow `make notice` to ensure that the lint checks can be satisfied
                                cmd(label: "make check-go", script: "make check-go")
```

**Review by cachedout — APPROVED** [review]
Nit: Add comment so we don't forget about this ordering in the future.

**Review by kuisathaverat — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (jsoriano) [inline Makefile:113]: If we run `go mod tidy` here we need to run also `go mod download` to be consistent with `make notice`. I have opened a PR for this: https://github.com/elastic/beats/pull/26241

**Pull request merged**